### PR TITLE
Remove .localized property inside the blog_index_feature template

### PR DESCRIPTION
**Pull request by AdalbertoMoz**
# Description

The following changes are related to removing the `.localized` and `.specific` properties inside the `blog_index_feature.html` template. 

The following elements inside the template have to be reviewed to confirm that it's working as expected:
- **Blog Title, Description and Topic** for both First Blog shown (on the Index Blog Page) and Featured Video Post.
- **Anchor Tag on Image** for First Blog shown (on the Index Blog Page).

Example of First Blog shown (on the Index Blog Page):
image

Example of Featured Video Post:
image

Link to sample test page: https://foundation-s-feature-10-z1rsj8.mofostaging.net/
Related PRs/issues: #10005

# To Test

1. Go to the [Review App Blog Index Page](https://foundation-s-feature-10-z1rsj8.mofostaging.net/en/blog/)
2. Confirm the following elements are working as expected for the **First Blog shown**:
    - **Blog Title, Description and Blog Image** properly redirect to the **Blog Page**
    - **Topic anchor** properly redirect to the **Blog Topics Template**
4. Confirm the following elements are working as expected for the **Featured Video Post**:
    - **Blog Title and Description** properly redirect to the **Blog Page**
    - **Topic anchor** properly redirect to the **Blog Topics Template**

# Checklist

~~**Tests**~~
~~- [ ] Is the code I'm adding covered by tests?~~

~~**Changes in Models:**~~
~~- [ ] Did I update or add new fake data?~~
~~- [ ] Did I squash my migration?~~
~~- [ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

~~**Documentation:**~~
~~- [ ] Is my code documented?~~
~~- [ ] Did I update the READMEs or wagtail documentation?~~

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-752)

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py [hunk 1]
```diff
@@ -510,7 +510,12 @@ def localised_featured_pages(self):
         featured_blog_pages = BlogPage.objects.filter(featured_pages_relationship__page=self).order_by(
             "featured_pages_relationship__sort_order"
         )
-        return localize_queryset(featured_blog_pages, preserve_order=True)
+
+        # Add .specific property to return latest published version
+        localised_featured_blog_pages = [
+            page.specific for page in localize_queryset(featured_blog_pages, preserve_order=True)
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py:516]:
```suggestion

        localised_featured_blog_pages = [
            page for page in localize_queryset(featured_blog_pages, preserve_order=True)
```

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py [hunk 2]
```diff
@@ -519,6 +524,7 @@ def localised_featured_video_post(self):
         if not featured_post:
             return None
 
-        featured_post.blog_page = featured_post.blog_page.localized
+        # Add .specific property to return latest published version
+        featured_post.blog_page = featured_post.blog_page.localized.specific
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py:528]:
```suggestion
        featured_post.blog_page = featured_post.blog_page.localized
```

**Review by danielfmiranda — CHANGES_REQUESTED** [review]
Hi @AdalbertoMoz!

Thank you for your work on this PR, it looks great! I've followed the instructions in the description and can confirm that the review app is working as expected.

However, I believe we can simplify the code further by removing the use of `.specific` entirely, as it appears it is no longer needed.

In Wagtail, the `.specific` method returns pages as their most specific subclass. For example, when using `page.get_ancestors()` or `page.get_children()`, the results are returned as a list of the base `Page` model. This prevents access to any subclass-specific methods or properties, such as a `BlogPage`'s `get_meta_image` method or `topics` property, both of which we are accessing in the updated template.
 
I have tested the `localised_featured_pages`and `localised_featured_video_post`methods  locally, and confirmed they both already return querysets of `BlogPage`s. Therefore, using `.specific` is unnecessary, and we can avoid a few additional DB queries by removing it. 👍 

Thanks again for your hard work!

**Comment by AdalbertoMoz** [comment]
Hey @danielfmiranda thank you for your comment related to the `specific` property, makes totally sense and removing it makes a small improvement that can add up in the long run!

Just deleted the changes related to the `blog_index.py` file to keep it unchanged.

Could you please review once again the PR and the Review app to confirm everything works as expected?

Again, thanks in advance for your great help! 🙌

**Review by danielfmiranda — APPROVED** [review]
Thanks @AdalbertoMoz! LGTM 🚀

**Comment by data-sync-user** [comment]
➤ Simon Acosta Torres commented:

PR has been merged.

**Pull request merged**